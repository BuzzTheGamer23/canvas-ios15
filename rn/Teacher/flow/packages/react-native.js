// @flow
export type PanResponderGestureState = {
  stateID: number,
  moveX: number,
  moveY: number,
  x0: number,
  y0: number,
  dx: number,
  dy: number,
  vx: number,
  vy: number,
}
