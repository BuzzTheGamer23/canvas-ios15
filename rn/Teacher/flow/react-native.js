//
// Copyright (C) 2017-present Instructure, Inc.
//
// This program is free software: you can redistribute it and/or modify
// it under the terms of the GNU General Public License as published by
// the Free Software Foundation, version 3 of the License.
//
// This program is distributed in the hope that it will be useful,
// but WITHOUT ANY WARRANTY; without even the implied warranty of
// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
// GNU General Public License for more details.
//
// You should have received a copy of the GNU General Public License
// along with this program.  If not, see <http://www.gnu.org/licenses/>.
//

// @flow
/* eslint-disable */

// define basic style prop types
// the internal ones are the best we have
import {
  type ____ViewStyleProp_Internal,
  type ____TextStyleProp_Internal,
  type ____ImageStyleProp_Internal,
} from 'react-native/Libraries/StyleSheet/StyleSheetTypes'

declare type ViewStyleProp = ____ViewStyleProp_Internal
declare type TextStyleProp = ____TextStyleProp_Internal
declare type ImageStyleProp = ____ImageStyleProp_Internal
