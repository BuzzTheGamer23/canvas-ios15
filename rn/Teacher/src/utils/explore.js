/* @flow */

function hasProp (object: any, key: string, value: any): boolean {
  if (object instanceof Object) {
    const props = object.props
    if (props && props instanceof Object) {
      for (var prop in props) {
        if (prop === key && props[prop] === value) {
          return true
        }
      }
    }
  }

  return false
}

function selectByProp (object: any, propKey: string, value: any): ?any {
  if (!(typeof object === 'object')) {
    return null
  }

  if (hasProp(object, propKey, value)) {
    return object
  }

  let result = null

  for (var key in object) {
    if (object[key] instanceof Array) {
      object[key].forEach((e) => {
        result = selectByProp(e, propKey, value)
        if (result) {
          return result
        }
      })
    }

    result = selectByProp(object[key], propKey, value)
    if (result) {
      return result
    }
  }

  return result
}

class ComponentExplorer {
  component: any

  constructor (component: any) {
    this.component = component
  }

  selectByProp (propKey: string, value: any): ?any {
    return selectByProp(this.component, propKey, value)
  }

  selectByID (id: string): ?any {
    return this.selectByProp('testID', id)
  }
}

export default function explore (component: any): ComponentExplorer {
  return new ComponentExplorer(component)
}
