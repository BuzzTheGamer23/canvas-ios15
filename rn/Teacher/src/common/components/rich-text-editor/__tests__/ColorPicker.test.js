/* @flow */

import 'react-native'
import React from 'react'
import renderer from 'react-test-renderer'

import ColorPicker from '../ColorPicker'
import explore from '../../../../../test/helpers/explore'

jest
  .mock('TouchableHighlight', () => 'TouchableHighlight')

describe('ColorPicker', () => {
  const defaultProps = {
    pickedColor: jest.fn(),
  }

  it('renders', () => {
    expect(
      renderer.create(
        <ColorPicker {...defaultProps} />
      ).toJSON()
    ).toMatchSnapshot()
  })

  it('triggers picked color', () => {
    const pickedColor = jest.fn()
    const tree = renderer.create(
      <ColorPicker {...defaultProps} pickedColor={pickedColor} />
    ).toJSON()

    const white: any = explore(tree).selectByID('color-picker-option-white')
    white.props.onPress()

    expect(pickedColor).toHaveBeenCalledWith('white')

    const black: any = explore(tree).selectByID('color-picker-option-black')
    black.props.onPress()

    expect(pickedColor).toHaveBeenCalledWith('black')
  })
})
