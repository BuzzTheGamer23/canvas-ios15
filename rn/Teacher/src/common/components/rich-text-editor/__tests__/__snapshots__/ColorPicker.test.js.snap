// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ColorPicker renders 1`] = `
<View
  style={
    Object {
      "backgroundColor": "white",
      "borderTopColor": "#C7CDD1",
      "borderTopWidth": 1,
    }
  }
>
  <RCTScrollView
    horizontal={true}
  >
    <View>
      <View
        style={
          Object {
            "alignItems": "center",
            "flexDirection": "row",
            "justifyContent": "center",
          }
        }
      >
        <TouchableHighlight
          onPress={[Function]}
          style={
            Object {
              "height": 44,
              "width": 44,
            }
          }
          testID="color-picker-option-white"
          underlayColor="transparent"
        >
          <View
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "borderRadius": 19,
                  "flex": 1,
                  "justifyContent": "center",
                  "margin": 6,
                },
                Object {
                  "backgroundColor": "white",
                  "borderColor": "#E6E9EA",
                  "borderWidth": 1,
                },
              ]
            }
          />
        </TouchableHighlight>
        <TouchableHighlight
          onPress={[Function]}
          style={
            Object {
              "height": 44,
              "width": 44,
            }
          }
          testID="color-picker-option-black"
          underlayColor="transparent"
        >
          <View
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "borderRadius": 19,
                  "flex": 1,
                  "justifyContent": "center",
                  "margin": 6,
                },
                Object {
                  "backgroundColor": "black",
                  "borderColor": "transparent",
                  "borderWidth": 0,
                },
              ]
            }
          />
        </TouchableHighlight>
        <TouchableHighlight
          onPress={[Function]}
          style={
            Object {
              "height": 44,
              "width": 44,
            }
          }
          testID="color-picker-option-#8B969E"
          underlayColor="transparent"
        >
          <View
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "borderRadius": 19,
                  "flex": 1,
                  "justifyContent": "center",
                  "margin": 6,
                },
                Object {
                  "backgroundColor": "#8B969E",
                  "borderColor": "transparent",
                  "borderWidth": 0,
                },
              ]
            }
          />
        </TouchableHighlight>
        <TouchableHighlight
          onPress={[Function]}
          style={
            Object {
              "height": 44,
              "width": 44,
            }
          }
          testID="color-picker-option-#EC463D"
          underlayColor="transparent"
        >
          <View
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "borderRadius": 19,
                  "flex": 1,
                  "justifyContent": "center",
                  "margin": 6,
                },
                Object {
                  "backgroundColor": "#EC463D",
                  "borderColor": "transparent",
                  "borderWidth": 0,
                },
              ]
            }
          />
        </TouchableHighlight>
        <TouchableHighlight
          onPress={[Function]}
          style={
            Object {
              "height": 44,
              "width": 44,
            }
          }
          testID="color-picker-option-#E42565"
          underlayColor="transparent"
        >
          <View
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "borderRadius": 19,
                  "flex": 1,
                  "justifyContent": "center",
                  "margin": 6,
                },
                Object {
                  "backgroundColor": "#E42565",
                  "borderColor": "transparent",
                  "borderWidth": 0,
                },
              ]
            }
          />
        </TouchableHighlight>
        <TouchableHighlight
          onPress={[Function]}
          style={
            Object {
              "height": 44,
              "width": 44,
            }
          }
          testID="color-picker-option-#8E4195"
          underlayColor="transparent"
        >
          <View
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "borderRadius": 19,
                  "flex": 1,
                  "justifyContent": "center",
                  "margin": 6,
                },
                Object {
                  "backgroundColor": "#8E4195",
                  "borderColor": "transparent",
                  "borderWidth": 0,
                },
              ]
            }
          />
        </TouchableHighlight>
        <TouchableHighlight
          onPress={[Function]}
          style={
            Object {
              "height": 44,
              "width": 44,
            }
          }
          testID="color-picker-option-#654C9B"
          underlayColor="transparent"
        >
          <View
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "borderRadius": 19,
                  "flex": 1,
                  "justifyContent": "center",
                  "margin": 6,
                },
                Object {
                  "backgroundColor": "#654C9B",
                  "borderColor": "transparent",
                  "borderWidth": 0,
                },
              ]
            }
          />
        </TouchableHighlight>
        <TouchableHighlight
          onPress={[Function]}
          style={
            Object {
              "height": 44,
              "width": 44,
            }
          }
          testID="color-picker-option-#4656A2"
          underlayColor="transparent"
        >
          <View
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "borderRadius": 19,
                  "flex": 1,
                  "justifyContent": "center",
                  "margin": 6,
                },
                Object {
                  "backgroundColor": "#4656A2",
                  "borderColor": "transparent",
                  "borderWidth": 0,
                },
              ]
            }
          />
        </TouchableHighlight>
        <TouchableHighlight
          onPress={[Function]}
          style={
            Object {
              "height": 44,
              "width": 44,
            }
          }
          testID="color-picker-option-#2684C2"
          underlayColor="transparent"
        >
          <View
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "borderRadius": 19,
                  "flex": 1,
                  "justifyContent": "center",
                  "margin": 6,
                },
                Object {
                  "backgroundColor": "#2684C2",
                  "borderColor": "transparent",
                  "borderWidth": 0,
                },
              ]
            }
          />
        </TouchableHighlight>
        <TouchableHighlight
          onPress={[Function]}
          style={
            Object {
              "height": 44,
              "width": 44,
            }
          }
          testID="color-picker-option-#3BA5DA"
          underlayColor="transparent"
        >
          <View
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "borderRadius": 19,
                  "flex": 1,
                  "justifyContent": "center",
                  "margin": 6,
                },
                Object {
                  "backgroundColor": "#3BA5DA",
                  "borderColor": "transparent",
                  "borderWidth": 0,
                },
              ]
            }
          />
        </TouchableHighlight>
        <TouchableHighlight
          onPress={[Function]}
          style={
            Object {
              "height": 44,
              "width": 44,
            }
          }
          testID="color-picker-option-#2EBCD1"
          underlayColor="transparent"
        >
          <View
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "borderRadius": 19,
                  "flex": 1,
                  "justifyContent": "center",
                  "margin": 6,
                },
                Object {
                  "backgroundColor": "#2EBCD1",
                  "borderColor": "transparent",
                  "borderWidth": 0,
                },
              ]
            }
          />
        </TouchableHighlight>
        <TouchableHighlight
          onPress={[Function]}
          style={
            Object {
              "height": 44,
              "width": 44,
            }
          }
          testID="color-picker-option-#219588"
          underlayColor="transparent"
        >
          <View
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "borderRadius": 19,
                  "flex": 1,
                  "justifyContent": "center",
                  "margin": 6,
                },
                Object {
                  "backgroundColor": "#219588",
                  "borderColor": "transparent",
                  "borderWidth": 0,
                },
              ]
            }
          />
        </TouchableHighlight>
        <TouchableHighlight
          onPress={[Function]}
          style={
            Object {
              "height": 44,
              "width": 44,
            }
          }
          testID="color-picker-option-#479F4B"
          underlayColor="transparent"
        >
          <View
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "borderRadius": 19,
                  "flex": 1,
                  "justifyContent": "center",
                  "margin": 6,
                },
                Object {
                  "backgroundColor": "#479F4B",
                  "borderColor": "transparent",
                  "borderWidth": 0,
                },
              ]
            }
          />
        </TouchableHighlight>
        <TouchableHighlight
          onPress={[Function]}
          style={
            Object {
              "height": 44,
              "width": 44,
            }
          }
          testID="color-picker-option-#8CC152"
          underlayColor="transparent"
        >
          <View
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "borderRadius": 19,
                  "flex": 1,
                  "justifyContent": "center",
                  "margin": 6,
                },
                Object {
                  "backgroundColor": "#8CC152",
                  "borderColor": "transparent",
                  "borderWidth": 0,
                },
              ]
            }
          />
        </TouchableHighlight>
        <TouchableHighlight
          onPress={[Function]}
          style={
            Object {
              "height": 44,
              "width": 44,
            }
          }
          testID="color-picker-option-#FBBF40"
          underlayColor="transparent"
        >
          <View
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "borderRadius": 19,
                  "flex": 1,
                  "justifyContent": "center",
                  "margin": 6,
                },
                Object {
                  "backgroundColor": "#FBBF40",
                  "borderColor": "transparent",
                  "borderWidth": 0,
                },
              ]
            }
          />
        </TouchableHighlight>
        <TouchableHighlight
          onPress={[Function]}
          style={
            Object {
              "height": 44,
              "width": 44,
            }
          }
          testID="color-picker-option-#F69637"
          underlayColor="transparent"
        >
          <View
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "borderRadius": 19,
                  "flex": 1,
                  "justifyContent": "center",
                  "margin": 6,
                },
                Object {
                  "backgroundColor": "#F69637",
                  "borderColor": "transparent",
                  "borderWidth": 0,
                },
              ]
            }
          />
        </TouchableHighlight>
        <TouchableHighlight
          onPress={[Function]}
          style={
            Object {
              "height": 44,
              "width": 44,
            }
          }
          testID="color-picker-option-#ED5A34"
          underlayColor="transparent"
        >
          <View
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "borderRadius": 19,
                  "flex": 1,
                  "justifyContent": "center",
                  "margin": 6,
                },
                Object {
                  "backgroundColor": "#ED5A34",
                  "borderColor": "transparent",
                  "borderWidth": 0,
                },
              ]
            }
          />
        </TouchableHighlight>
        <TouchableHighlight
          onPress={[Function]}
          style={
            Object {
              "height": 44,
              "width": 44,
            }
          }
          testID="color-picker-option-#EE6491"
          underlayColor="transparent"
        >
          <View
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "borderRadius": 19,
                  "flex": 1,
                  "justifyContent": "center",
                  "margin": 6,
                },
                Object {
                  "backgroundColor": "#EE6491",
                  "borderColor": "transparent",
                  "borderWidth": 0,
                },
              ]
            }
          />
        </TouchableHighlight>
      </View>
    </View>
  </RCTScrollView>
</View>
`;
